lms/views/lti/basic_launch.py

```diff
@@ -16,24 +16,37 @@
 
 from lms.models import LtiLaunches
 from lms.security import Permissions
-from lms.services import LTIRoleService
+from lms.services import DocumentURLService, LTIRoleService
 from lms.services.assignment import AssignmentService
-from lms.services.vitalsource.client import VitalSourceService
-from lms.validation import (
-    BasicLTILaunchSchema,
-    ConfigureAssignmentSchema,
-    URLConfiguredBasicLTILaunchSchema,
-)
+from lms.validation import BasicLTILaunchSchema, ConfigureAssignmentSchema
 from lms.validation.authentication import BearerTokenSchema
 
 
-class ResourceLinkParam:
-    # A normal LTI (non-deep linked) launch
-    LTI = "resource_link_id"
-    # A Brightspace course we can copy
-    COPIED_BRIGHTSPACE = "ext_d2l_resource_link_id_history"
-    # A Blackboard course we can copy
-    COPIED_BLACKBOARD = "resource_link_id_history"
+def has_document_url(context, request):
+    """Get if the current launch has a resolvable document URL."""
+    return bool(
+        request.find_service(DocumentURLService).get_document_url(context, request)
+    )
+
+
+def is_authorized_to_configure_assignments(_context, request):
+    """Get if the current user allowed to configured assignments."""
+
+    if not request.lti_user:
+        return False
+
+    roles = request.lti_user.roles.lower()
+
+    return any(
+        role in roles for role in ["administrator", "instructor", "teachingassistant"]
+    )
+
+
+# This is imported in `lms.views.predicates`
+LTI_LAUNCH_PREDICATES = {
+    "has_document_url": has_document_url,
+    "authorized_to_configure_assignments": is_authorized_to_configure_assignments,
+}
```

Comment:
These are the predicates now, and they are here because it's the only place they are used